### ISSUE TITLE
Problem with nested regex groups

### ISSUE THREAD

**Issue by fiedlerp** [ponyisi/histogram_postprocessing#13]
When I try to match histogram with regex containing nested groups, the histgrinder crashes. There is a workaround but it is semantically different.

Example from AFP: `(?P<station>(far|near)[AC]side)`
Workaround: `(?P<station>[far|near]*[AC]side)`

Example from TRT: `(?P<region>(EndcapA|EndcapC)/Sector\d.)`
Workaround: `(?P<region>[EndcapA|EndcapC]*/Sector\d.)`

The problem is that there cannot be nested named groups like: `(?P<part>(?=Front|Middle|End)(?P<part_letter>[FME])(ront|iddle|nd))`

**Comment by ponyisi** [comment]
Hi @Tucnace ,

Could you provide the full snippet from the YAML file that causes the crash, as well as the backtrace that you get from the crash?

Best,
Peter

**Comment by fiedlerp** [comment]
Hi @ponyisi,

here is the YAML snippet. I have tested that this is enough to crash it.
```yaml
Input: [ 'AFP/SiT/TracksVsLb/lbTracksAll_(?P<station>(far|near)[AC]side)', 'AFP/SiT/Aux/NumberOfEventsPerLumiblock' ]
Output: [ 'AFP/SiT/PP/TrackFME/lbTracksAll_vs_lb_{station}' ]
Function: histgrinder.example.transform_function_divide_ROOT
Description: TrackAll
```

And here is the stack trace:
```stacktrace
Traceback (most recent call last):
  File "/cvmfs/atlas-nightlies.cern.ch/repo/sw/master_Athena_x86_64-centos7-gcc11-opt/sw/lcg/releases/LCG_101_ATLAS_20/Python/3.9.6/x86_64-centos7-gcc11-opt/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/cvmfs/atlas-nightlies.cern.ch/repo/sw/master_Athena_x86_64-centos7-gcc11-opt/sw/lcg/releases/LCG_101_ATLAS_20/Python/3.9.6/x86_64-centos7-gcc11-opt/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/cvmfs/atlas-nightlies.cern.ch/repo/sw/master_Athena_x86_64-centos7-gcc11-opt/2022-05-11T2101/AthenaExternals/22.0.67/InstallArea/x86_64-centos7-gcc11-opt/lib/python3.9/site-packages/histgrinder/engine.py", line 93, in <module>
    go()
  File "/cvmfs/atlas-nightlies.cern.ch/repo/sw/master_Athena_x86_64-centos7-gcc11-opt/2022-05-11T2101/AthenaExternals/22.0.67/InstallArea/x86_64-centos7-gcc11-opt/lib/python3.9/site-packages/histgrinder/engine.py", line 66, in go
    _.consider(obj)
  File "/cvmfs/atlas-nightlies.cern.ch/repo/sw/master_Athena_x86_64-centos7-gcc11-opt/2022-05-11T2101/AthenaExternals/22.0.67/InstallArea/x86_64-centos7-gcc11-opt/lib/python3.9/site-packages/histgrinder/transform.py", line 59, in consider
    return self.transform()
  File "/cvmfs/atlas-nightlies.cern.ch/repo/sw/master_Athena_x86_64-centos7-gcc11-opt/2022-05-11T2101/AthenaExternals/22.0.67/InstallArea/x86_64-centos7-gcc11-opt/lib/python3.9/site-packages/histgrinder/transform.py", line 73, in transform
    groupedfirstmatches = self._groupMatches(firstmatches)
  File "/cvmfs/atlas-nightlies.cern.ch/repo/sw/master_Athena_x86_64-centos7-gcc11-opt/2022-05-11T2101/AthenaExternals/22.0.67/InstallArea/x86_64-centos7-gcc11-opt/lib/python3.9/site-packages/histgrinder/transform.py", line 117, in _groupMatches
    reducedtup = tuple(v for i, v in enumerate(tup)
  File "/cvmfs/atlas-nightlies.cern.ch/repo/sw/master_Athena_x86_64-centos7-gcc11-opt/2022-05-11T2101/AthenaExternals/22.0.67/InstallArea/x86_64-centos7-gcc11-opt/lib/python3.9/site-packages/histgrinder/transform.py", line 118, in <genexpr>
    if self.regextupnames[0][i] in self.outputnames)
IndexError: tuple index out of range
```

Cheers,
Petr

**Comment by ponyisi** [comment]
Fixed in 0.1.6 which is in Athena nightlies after 2022-05-17T2101, closing.
